### PR TITLE
Unify settings access on droid

### LINKED ISSUE #5931: Calendar: Non-confusing settings button
Settings button will be in the top right corner of the calendar view.

- [x] corner view of calendar
- [x] corner view of main log (like iOS)
- [x] remove tab bar settings item

It leads to general settings (not calendar settings any more)

(https://app.zeplin.io/project/5a0d3cd27a75add1a8924f6e/screen/5a8613f5a212ab0471a3e0b4) positioning if easily possible

### PR DESCRIPTION
**Pull request by daividssilverio**
## What's this?
This PR does 2 things:
1. Removes the settings tab from the main activity bottom navigation bar
2. Adds the settings menu (with the gear icon) to all main fragments (main log, reports, calendar).

### Relationships
Closes #5931 

## How is it done?
The settings fragment is now an activity (once again).
The toolbars handle clicks on the inner toolbars that will navigate to the `SettingsActivity`.

## :squid: Permissions
🦑 📆 pals

### CODE REVIEW

File: Toggl.Core.Tests/UI/ViewModels/MainTabBarViewModelTests.cs [hunk 1]
```diff
@@ -118,15 +118,15 @@ public sealed class TheConstructor : MainTabViewModelTest
         public sealed class TheTabsProperty : MainTabViewModelTest
         {
             [Theory]
-            [InlineData(Platform.Daneel, false)]
-            [InlineData(Platform.Giskard, true)]
-            public void ShouldContainTheSettingsViewModelBasedOntheRemoteConfigService(Platform platform, bool includesSettings)
+            [InlineData(Platform.Daneel)]
+            [InlineData(Platform.Giskard)]
+            public void ShouldContainTheSettingsViewModelBasedOntheRemoteConfigService(Platform platform)
```
Comment (heytherewill) [inline Toggl.Core.Tests/UI/ViewModels/MainTabBarViewModelTests.cs]: This test has no reason to exist

File: Toggl.Core.UI/ViewModels/MainTabBarViewModel.cs [hunk 2]
```diff
@@ -149,11 +136,6 @@ private IEnumerable<ViewModel> getViewModels()
             yield return mainViewModel;
             yield return reportsViewModel;
             yield return calendarViewModel;
-
-            if (platformInfo.Platform == Platform.Giskard)
```
Comment (heytherewill) [inline Toggl.Core.UI/ViewModels/MainTabBarViewModel.cs]: Do we still need to inject `platformInfo`?

File: Toggl.Droid/Activities/SettingsActivity.cs [hunk 3]
```diff
@@ -207,16 +208,11 @@ private void setSyncStatusView(PresentableSyncStatus status)
...
-        public void ScrollToStart()
-        {
-            scrollView?.SmoothScrollTo(0, 0);
-        }
-
-        private void showFeedbackSuccessToast(bool succeeeded)
+        private void showFeedbackSuccessToast(bool succeeded)
         {
-            if (!succeeeded) return;
+            if (!succeeded) return;
```
Comment (heytherewill) [inline Toggl.Droid/Activities/SettingsActivity.cs]: Needs MORE E

Comment (heytherewill) [reply Toggl.Droid/Activities/SettingsActivity.cs]: image

File: Toggl.Droid/Presentation/ActivityPresenter.cs [hunk 4]
```diff
@@ -55,7 +56,8 @@ public sealed class ActivityPresenter : AndroidPresenter
             [typeof(SendFeedbackViewModel)] = new ActivityPresenterInfo(typeof(SendFeedbackActivity)),
             [typeof(SignupViewModel)] = new ActivityPresenterInfo(typeof(SignUpActivity), clearBackStackFlags),
             [typeof(StartTimeEntryViewModel)] = new ActivityPresenterInfo(typeof(StartTimeEntryActivity)),
-            [typeof(TokenResetViewModel)] = new ActivityPresenterInfo(typeof(TokenResetActivity), clearBackStackFlags)
+            [typeof(TokenResetViewModel)] = new ActivityPresenterInfo(typeof(TokenResetActivity), clearBackStackFlags),
+            [typeof(SettingsViewModel)] = new ActivityPresenterInfo(typeof(SettingsActivity))
```
Comment (heytherewill) [inline Toggl.Droid/Presentation/ActivityPresenter.cs]: Alphabetical order please 🙈

File: Toggl.Core.Tests/UI/ViewModels/MainTabBarViewModelTests.cs (hunk 1 above)
Comment (daividssilverio) [reply Toggl.Core.Tests/UI/ViewModels/MainTabBarViewModelTests.cs]: 🔥

File: Toggl.Droid/Activities/SettingsActivity.cs (hunk 3 above)
Comment (daividssilverio) [reply Toggl.Droid/Activities/SettingsActivity.cs]: reverting.

File: Toggl.Droid/Presentation/ActivityPresenter.cs (hunk 4 above)
Comment (daividssilverio) [reply Toggl.Droid/Presentation/ActivityPresenter.cs]: 🤔 We should really put this on some docs if it's not already.
This and other 💄 that we ask on PRs.

File: Toggl.Core.UI/ViewModels/MainTabBarViewModel.cs (hunk 2 above)
Comment (daividssilverio) [reply Toggl.Core.UI/ViewModels/MainTabBarViewModel.cs]: wait, you reviewed on an intermediate commit.

Comment (daividssilverio) [reply Toggl.Core.UI/ViewModels/MainTabBarViewModel.cs]: Wait, I understood the wrong question. But yeah, I don't think we need to inject it.

Comment (daividssilverio) [reply Toggl.Core.UI/ViewModels/MainTabBarViewModel.cs]: I need more coffee... Those are being used in the `MainViewModel`.

**Pull request merged**